### DISCUSSION TITLE
Notification [Upgrade] : App upgraded 🚀 Please clear the cache: => Always shown :/

### DISCUSSION THREAD

**Discussion by ytorres** [netalertx/NetAlertX#908] [General]
Hi all and thanks a lot for this great app !

On first install, I have always "Notification [Upgrade] : App upgraded 🚀 Please clear the cache: blabla" popup shown :/

What am I doing wrong ?

Best

**Comment by jokob-sk** [comment]
If this always displays and doesn't disappear then usually the backend restarts for some reason. Inspect the logs to see if something is causing backend restarts. 

Similar closed issues: 

- https://github.com/jokob-sk/NetAlertX/issues/811
- https://github.com/jokob-sk/NetAlertX/issues/791

**Reply by ytorres** [reply]
I see nothing special in app.log...:/
[app.log](https://github.com/user-attachments/files/18044189/app.log)

**Reply by jokob-sk** [reply]
It seems your app is not restarting, which is good, but the weird thing is you are getting command not found errors, which I would not expect in docker.

`20:25:15 [Plugins] Output: sudo: arp-scan : commande introuvable`

Could you please let me know what docker compose file you are using or how did you deploy the app?

The message is to be designed to show up if you upgrade the application, so that's by design. 

Also - thanks a lot helping translating 🙏

**Reply by ytorres** [reply]
I use a fresh debian bookworm install on a virtual machine.

I just add "arp-scan" but the result is the same...

I suggest to add arp-scan into the dependencies install script ..! ;)

**Comment by ytorres** [comment]
I found a "Permission denied" in nginx error log file :


`2024/12/07 09:32:46 [error] 824#824: *13187 FastCGI sent in stderr: "PHP message: PHP Warning:  file_put_contents(/app/front/api/user_notifications.json): Failed to open stream: Permission denied in /app/front/php/server/utilNotification.php on line 151" while reading response header from upstream, client: 192.168.1.102, server: , request: "GET /php/server/utilNotification.php?action=mark_notification_as_read&guid=a4f17b62-b7a5-4512-add0-c18e2cb741c0 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php8.2-fpm.sock:", host: "192.168.1.105:20211", referrer: "http://192.168.1.105:20211/devices.php"`

All files in /app/front/api/ are "root:root". I fix "user_notifications.json" to "root:www-data" ; this error has dissapered but always the same notification problem

**Reply by ytorres** [reply]
On the first load of /index.php => devices.php, i found this :

```
URL de requête:
http://192.168.1.105:20211/php/server/query_graphql.php
Méthode de requête:
POST
Code d'état:
500 Internal Server Error
```
Somethings went wrong with graphql but i can't found where for now...

**Comment by ytorres** [comment]
I can suggest to add openrc to the dependecies script too. openrc is not install by default with Debian Bookworm.

I continue to search why graphql return a 500 internal error...

**Reply by jokob-sk** [reply]
Humm, you can usually find these errors in these 2 files:

`cat /var/log/nginx/error.log`

`cat /app/log/app.php_errors.log`

**Reply by jokob-sk** [reply]
Also, make sure you use the latest code base - I'm working on a lot of things around there as well - refactoring for performance and security

**Reply by jokob-sk** [reply]
`openrc` is being installed here: https://github.com/jokob-sk/NetAlertX/blob/74df6601457fe01f74e8e0de9ee5ebd92cd5f45d/install/install_dependencies.debian.sh#L19 , not sure why it's missing

**Reply by ytorres** [reply]
Ok, this is the problem :

`````# Install dependencies`

apt-get install -y  \
    tini snmp ca-certificates curl libwww-perl arp-scan perl apt-utils cron sudo \
    nginx-light php php-cgi php-fpm php-sqlite3 php-curl php-openssl sqlite3 dnsutils net-tools \
    python3 python3-dev iproute2 nmap python3-pip zip systemctl usbutils traceroute nbtscan avahi avahi-tools openrc dbus build-essential git``

In Debian bookworm, this packages are not longer available :

- php-openssl
- avahi
- avahi-tools

so this apt-get failed and no more packages are installed (the lack of php-curl cause the error 500 with graphql, for example & this example why I have to install openrc manually)

We should find packages replacment for this 3 packages and update the script

**Reply by ytorres** [reply]
1/ According to https://packages.debian.org/bookworm/php8.2-fpm : php-openssl is embed with php8.2-fpm ; you can just remove it from the script

2/ Replace avahi by avahi-daemon

3/ Replace avahi-tools by avahi-utils

4/ Remove dbus

This should work on a fresh install. I will give it a try and tell you ;)
